src/components/Banners/SlackNotificationBanner.tsx

```diff
@@ -77,18 +33,39 @@ export const SlackNotificationBannerCore = () => {
   >(UPDATE_USER_SETTINGS, {
     onCompleted: () => {
       hideBanner();
-      dispatchBanner.successBanner(
+      dispatchToast.success(
         "You will now receive Slack notifications when your patches fail or succeed"
       );
     },
     onError: (err) => {
-      dispatchBanner.errorBanner(
-        `Error while saving settings: '${err.message}'`
-      );
+      dispatchToast.error(`Error while saving settings: '${err.message}'`);
     },
     refetchQueries: ["GetUserSettings"],
   });
 
+  // USER SETTINGS QUERY
+  const { data: userSettingsData } = useQuery<GetUserSettingsQuery>(
+    GET_USER_SETTINGS
+  );
+  const { userSettings } = userSettingsData || {};
+  const { slackUsername: defaultSlackUsername, notifications } =
+    userSettings || {};
+  const { patchFinish, patchFirstFailure } = notifications || {};
+
+  const [slackUsername, setSlackUsername] = useState(
+    () => defaultSlackUsername
```

Comment:
`() =>` seems like a typo

Reply:
It is not a typo passing a function into useState allows you to lazily init the state. Doing this basically says set the initial state when defaultSlackUsername becomes defined. Allows us to avoid a useEffect to update the state when `defaultSlackUsername` becomes defined.

Reply:
https://kentcdodds.com/blog/use-state-lazy-initialization-and-function-updates/